src/lib/lspace/elements/border_element.rs

```diff
@@ -0,0 +1,186 @@
+use cairo::Context;
+
+use std::rc::Rc;
+use std::cell::{RefCell, Ref, RefMut};
+
+use layout::lalloc::LAlloc;
+use layout::lreq::LReq;
+use geom::bbox2::BBox2;
+use graphics::border::Border;
+use elements::element_layout::{ElementReq, ElementAlloc};
+use elements::element::{TElement, ElementRef, ElementParentMut};
+use elements::container::TContainerElement;
+use elements::bin::{TBinElement, BinComponentMut};
+use elements::container_sequence::{TContainerSequenceElement};
+use elements::root_element::{TRootElement};
+
+
+struct BorderElementMut {
+    parent: ElementParentMut,
+    req: ElementReq,
+    alloc: ElementAlloc,
+    bin: BinComponentMut,
+}
+
+pub struct BorderElement {
+    m: RefCell<BorderElementMut>,
+    border: Rc<Border>,
+}
+
+impl BorderElement {
+    pub fn new(border: &Rc<Border>) -> BorderElement {
+        return BorderElement{m: RefCell::new(BorderElementMut{
+            parent: ElementParentMut::new(),
+            req: ElementReq::new(), alloc: ElementAlloc::new(),
+            bin: BinComponentMut::new()}),
+            border: border.clone()};
+    }
+}
+
+impl TElement for BorderElement {
+    /// Interface acquisition
+    fn as_container(&self) -> Option<&TContainerElement> {
+        return Some(self);
+    }
+
+    fn as_bin(&self) -> Option<&TBinElement> {
+        return Some(self);
+    }
+
+    fn as_container_sequence(&self) -> Option<&TContainerSequenceElement> {
+        return None
+    }
+
+    fn as_root_element(&self) -> Option<&TRootElement> {
+        return None;
+    }
+
+    /// Parent get and set methods
+    fn get_parent(&self) -> Option<ElementRef> {
+        return self.m.borrow().parent.get().clone();
+    }
+
+    fn set_parent(&self, p: Option<&ElementRef>) {
+        self.m.borrow_mut().parent.set(p);
+    }
+
+    // Layout structure acquisition
+    fn element_req(&self) -> Ref<ElementReq> {
+        return Ref::map(self.m.borrow(), |m| &m.req);
+    }
+
+    fn element_alloc(&self) -> Ref<ElementAlloc> {
+        return Ref::map(self.m.borrow(), |m| &m.alloc);
+    }
+
+    fn element_alloc_mut(&self) -> RefMut<ElementAlloc> {
+        return RefMut::map(self.m.borrow_mut(), |m| &mut m.alloc);
+    }
+
+    /// Update element X requisition
+    fn element_update_x_req(&self, x_req: &LReq) -> bool {
+        return self.m.borrow_mut().req.update_x_req(x_req);
+    }
+
+    /// Update element Y requisition
+    fn element_update_y_req(&self, y_req: &LReq) -> bool {
+        return self.m.borrow_mut().req.update_y_req(y_req);
+    }
+
+    /// Paint the element content that is contributed by the element itself
+    fn draw_self(&self, cairo_ctx: &Context, visible_region: &BBox2) {
+        let mm = self.m.borrow();
+        let alloc = &mm.alloc;
+        self.border.draw_background(cairo_ctx,
```

Comment:
The indent of this call's parameters doesn't match the call immediately below it.